core/imt/src/TPoolManager.cxx

```diff
@@ -35,21 +31,16 @@ namespace ROOT {
          // running when the constructor was called.
          if (mustDelete) {
             fSched->terminate();
-            fgPoolSize = 0;
          }
       }
 
-      //Number of threads the PoolManager has been initialized with.
-      UInt_t TPoolManager::GetPoolSize()
-      {
-         return fgPoolSize;
-      }
-
       //Factory function returning a shared pointer to the only instance of the PoolManager.
-      std::shared_ptr<TPoolManager> GetPoolManager(UInt_t nThreads)
+      std::shared_ptr<TPoolManager> GetPoolManager()
       {
          if (GetWP().expired()) {
-            std::shared_ptr<TPoolManager> shared(new TPoolManager(nThreads));
+            // why not make_shared: https://stackoverflow.com/posts/20895705/
+            //(under "Why do instances of weak_ptrs keep the control block alive?"")
+            std::shared_ptr<TPoolManager> shared(new TPoolManager());
```

Comment:
I get a 404 for the link above. Consider adding the explanation here directly as to why you're not using make_shared, since that's a C++11 thing.

Reply:
TPoolManager works as a singleton that is referenced from several parts in ROOT (IMT, TThreadExecutor, TTreeProcessorMT...). We manage it's lifetime with a shared pointer, which provides automatic bookkeeping and allows us to destroy the scheduler when no one is holding a pointer to it. Destroying the scheduler is necessary, for instance, to change the number of threads.

Reply:
I will look about the comment or fix the link, thx